### PR TITLE
Add xml for domcapabilities for getting gic version

### PR DESCRIPTION
**Pull request by kylazhang**

### CODE REVIEW

File: virttest/libvirt_xml/domcapability_xml.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        accessors.XMLElementList(property_name='values',
+                                 libvirtxml=self,
+                                 parent_xpath='/',
+                                 marshal_from=self.marshal_from_values,
+                                 marshal_to=self.marshal_to_values)
+        super(EnumXML, self).__init__(virsh_instance=virsh_instance)
+        self.xml = '<enum/>'
+
+
+    @staticmethod
```
Comment (kylazhang) [inline virttest/libvirt_xml/domcapability_xml.py]: Thanks @balamuruhans , will fix all them soon

File: virttest/libvirt_xml/domcapability_xml.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
+"""
+Module simplifying manipulation of XML described at
+http://libvirt.org/formatcaps.html
```
Comment (waynesun09) [inline virttest/libvirt_xml/domcapability_xml.py]: http://libvirt.org/formatdomaincaps.html

Comment (kylazhang) [inline virttest/libvirt_xml/domcapability_xml.py]: Thanks, so careful, will update soon

File: virttest/libvirt_xml/domcapability_xml.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+from virttest import xml_utils
+from virttest.libvirt_xml import base, accessors, xcepts
+
+
+class DomCapabilityXML(base.LibvirtXMLBase):
+
+    """
+    Handler of libvirt domcapabilities operations.
+
```
Comment (will-Do) [inline virttest/libvirt_xml/domcapability_xml.py]: redundant blank line

File: virttest/libvirt_xml/domcapability_xml.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+                                 subclass=DomCapFeaturesXML,
+                                 subclass_dargs={
+                                     'virsh_instance': virsh_instance})
+
+        super(DomCapabilityXML, self).__init__(virsh_instance)
+        self['xml'] = self.__dict_get__('virsh').domcapabilities().stdout.strip()
+
+
+class DomCapFeaturesXML(base.LibvirtXMLBase):
+    """
```
Comment (will-Do) [inline virttest/libvirt_xml/domcapability_xml.py]: One blank line before the docstrings

File: virttest/libvirt_xml/domcapability_xml.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    Handler of feature element in libvirt domcapabilities.
+
```
Comment (will-Do) [inline virttest/libvirt_xml/domcapability_xml.py]: And here, it's better to add the 'Properties' here, you can check capability_xml.py

File: virttest/libvirt_xml/domcapability_xml.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+        for enum_node in self.xmltreefile.findall('/gic/enum'):
+            xml_str = xml_utils.ElementTree.tostring(enum_node)
+            new_enum = EnumXML()
+            new_enum.xml = xml_str
+            enum_list.append(new_enum)
+        return enum_list
+
+
+class ValueXML(base.LibvirtXMLBase):
+    """
```
Comment (will-Do) [inline virttest/libvirt_xml/domcapability_xml.py]: Same too

File: virttest/libvirt_xml/domcapability_xml.py [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+                                 tag_name='value')
+        super(ValueXML, self).__init__(virsh_instance=virsh_instance)
+        self.xml = '<value/>'
+
+
+class EnumXML(base.LibvirtXMLBase):
+    """
+    Handler of Enum element in libvirt domcapabilities
+
+    """
```
Comment (will-Do) [inline virttest/libvirt_xml/domcapability_xml.py]: this one

**Comment by kylazhang** [comment]
@will-Do Fix issues you mentioned and please help to check ,thanks

**Comment by waynesun09** [comment]
ACK

**Pull request merged**